src/EVM/Fetch.hs

```diff
@@ -186,12 +186,12 @@ fetchChainIdFrom url = do
 
 http :: Natural -> Maybe Natural -> BlockNumber -> Text -> Fetcher t m s
 http smtjobs smttimeout n url q =
-  withSolvers Z3 smtjobs smttimeout $ \s ->
+  withSolvers Z3 smtjobs 1 smttimeout $ \s ->
     oracle s (Just (n, url)) q
 
 zero :: Natural -> Maybe Natural -> Fetcher t m s
 zero smtjobs smttimeout q =
-  withSolvers Z3 smtjobs smttimeout $ \s ->
+  withSolvers Z3 smtjobs 1 smttimeout $ \s ->
```

Comment:
Fine to leave it as is for now, but maybe worth collecting these into some `SolverOpts` struct at some point?

Reply:
Yes, I agree. It's getting unwieldy. If we look at the changes I made to `test.hs`, it's clear we needed a helper function or it would have been a mess. Let me merge, and then I'll do a cleanup PR to fix!